lib/class-wp-rest-gutenberg-blocks.php

```diff
@@ -0,0 +1,374 @@
+	 *
+	 * @since 1.1.0
+	 *
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'blocks';
+	}
+
+	/**
+	 * Register the block routes.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		// Add a post collection block route.
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		$schema = $this->get_item_schema();
+		$get_item_args = array(
+			'context'  => $this->get_context_param( array( 'default' => 'view' ) ),
+		);
+
+		// Add a single post block route.
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\d]+)', array(
+			'args' => array(
+				'id' => array(
+					'description' => __( 'Unique identifier for the post.' ),
+					'type'        => 'integer',
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => $get_item_args,
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Checks if a given request has access to read posts and their blocks.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+
+		return true;
+	}
+
+	/**
+	 * Retrieves a collection of posts blocks.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+
+		// Retrieve the list of registered collection query parameters.
+		$registered = $this->get_collection_params();
+		$args = array();
+
+		/**
+		 * Filters the query arguments for a request.
+		 *
+		 * Enables adding extra arguments or setting defaults for a post blocks collection request.
+		 *
+		 * @since 1.1.0
+		 *
+		 * @link https://developer.wordpress.org/reference/classes/wp_query/
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args = apply_filters( "rest_blocks_query", $args, $request );
+		$query_args = $this->prepare_items_query( $args, $request );
+
+		$posts_query  = new WP_Query();
+		$query_result = $posts_query->query( $query_args );
+		$registry = WP_Block_Type_Registry::get_instance();
```

Comment:
Is the block type registry exposed anywhere? How do you get access to the available block types?

IMO, we should expose that at `/posts/blocks` to match `/posts/types` and `/posts/statuses`

Reply:
out of scope for this ticket